astropy/io/ascii/core.py

```diff
@@ -1463,7 +1470,8 @@ def _get_writer(Writer, fast_writer, **kwargs):
     if 'exclude_names' in kwargs:
         writer.exclude_names = kwargs['exclude_names']
     if 'fill_values' in kwargs:
-        writer.data.fill_values = kwargs['fill_values']
+        # Prepend user-specified values to the class default.  Both are lists.
+        writer.data.fill_values = kwargs['fill_values'] + writer.data.fill_values
```

Comment:
According to the docs `fill_values` for `write()` needs to be a list of tuples, but there has been a convenience built in to the reader-side that if there was only one tuple then you don't need the list around it.  It looks like that short-cut is built into the tests.  Urgh.